### PR TITLE
MOSIP-36423 - Automated Negative Scenarios for PMS

### PR DESCRIPTION
**Pull request by prathmeshj12**
Automated below APIs for Negative Scenarios PMS,

1. Deactivate Device
2. Deactivate SBI With Associated Devices 
3. Get All Approved Device Provider Ids
4. Get All Device List Mapped With Sbi Negative Scenarios

### CODE REVIEW

**Review by mohanachandran-s — APPROVED** [review]

File: api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
+DeactivateDevice:
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml:1]: Update this as per the xml file

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]

File: api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml:1]: Add "Neg", at the end of all negative testcases and also update the expectation in each test description

File: api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+      uniqueIdentifier: TC_PMS_DeactivateSBIWithAssociatedDevices_14
+      description: Deactivate SBI while keeping Request Time parameter as invalid
+      role: partnerrevampdevice
+      checkErrorsOnlyInResponse: true
+      restMethod: post
+      inputTemplate: pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios
+      outputTemplate: pms/error
+      input: '{
+      "id": "mosip.pms.deactivate.sbi.post",
+	  "version": "1.0",
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml:293]: correct the indentation

File: api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+      uniqueIdentifier: TC_PMS_DeactivateSBIWithAssociatedDevices_15
+      description: Deactivate SBI while keeping Object parameter as Missing
+      role: partnerrevampdevice
+      checkErrorsOnlyInResponse: true
+      restMethod: post
+      inputTemplate: pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios
+      outputTemplate: pms/error
+      input: '{
+      "id": "mosip.pms.deactivate.sbi.post",
+	  "version": "1.0",
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml:315]: correct the indentation

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]

File: api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml:1]: corrected the indentation

File: api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml (hunk 3 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml]: corrected the indentation

File: api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml (hunk 2 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/pms/DeactivateSBIWithAssociatedDevices/DeactivateSBIWithAssociatedDevicesNegativeScenarios.yml]: corrected the indentation

File: api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml
Comment (prathmeshj12) [reply api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml:1]: corrected the indentation and added Neg

File: api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml (hunk 1 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/pms/DeactivateDevice/DeactivateDeviceNegativeScenarios.yml]: Updated as per xml file

**Review by mohanachandran-s — APPROVED** [review]

**Review by mohanachandran-s — APPROVED** [review]

**Review by mayuradesh — APPROVED** [review]

**Pull request merged**